src/client/chowly-client.api.ts

```diff
@@ -0,0 +1,53 @@
+import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
+import { plainToClass } from 'class-transformer';
+import { IConfiguration } from '../interface';
+import { Sections } from '../model/chowly.model';
+import { sleep } from '../utils';
+
+export class ChowlyClient {
+    private readonly client: AxiosInstance;
+    private readonly apiKey: string;
+
+    private basePrefix: string = '/api/toos/v2';
+
+    constructor(private readonly config: IConfiguration) {
+        this.config = {
+            maxRetries: 3,
+            retryDelay: 2000,
+            ...config,
+        };
+        this.apiKey = this.config['api-key'];
+        this.client = axios.create({
+            baseURL: config.baseUrl,
+        });
+
+        this.client.interceptors.request.use(
+            async (config: AxiosRequestConfig): Promise<AxiosRequestConfig> => {
+                config.headers['api-key'] = this.apiKey;
+                return config;
+            },
+        );
+
+        this.client.interceptors.response.use(
+            undefined,
+            async (error): Promise<any> => {
+                if (!error.config.metadata) {
+                    error.config.metadata = { retryCount: 1 };
+                }
+                console.log('err', error);
+                if (error.response?.status >= 500 && error.config.metadata.retryCount < this.config.maxRetries!) {
+                    error.config.metadata.retryCount++;
+                    await sleep(this.config.retryDelay! + this.config.retryDelay! * 0.2 * Math.random());
```

Comment:
why you need `Math.random()`?

Reply:
some sort of delay, still thinking how to implement this correctly, or to use exponentialDelay from google, or to make something like this with given milliseconds + some kind of random